src/defaultsettings.cpp

```diff
@@ -76,13 +76,6 @@ static bool detect_touch()
 		return false;
 	}
 
-	return false;
-#elif defined(_WIN32)
-	// 0x01 The device has an integrated touch digitizer
-	// 0x80 The device is ready to receive digitizer input.
-	if ((GetSystemMetrics(SM_DIGITIZER) & 0x81) == 0x81)
```

Comment:
this is half-baked.
if we don't want to keep the auto-detection code we should delete it all.
if we still want it, this should just be guarded by `USE_SDL2` for the moment

Reply:
This code is still flawed even with the SDL2 device as it's checking for the existence of a touchscreen, not the form factor

Reply:
then delete it with that justification

Reply:
only the Windows code has this flaw and it is deleted

Reply:
that's what I meant.

Reply:
FWIW, I attempted to fix this by detecting KB + Mouse, but even after disconnecting either, the function does still return `true`.

```C
#include <windows.h>
#include <stdio.h>

bool have_keyboard_n_mouse() {
	bool have_keyboard = false;
	bool have_mouse = false;

	UINT devices_len;
	UINT ret = GetRawInputDeviceList(NULL, &devices_len, sizeof(RAWINPUTDEVICELIST));
	if (ret == 0) {
		RAWINPUTDEVICELIST *devices = (RAWINPUTDEVICELIST *)calloc(devices_len, sizeof(RAWINPUTDEVICELIST));
		ret = GetRawInputDeviceList(devices, &devices_len, sizeof(RAWINPUTDEVICELIST));

		if (devices && ret != (UINT)-1) {
			printf("devs: %d\n", devices_len);
			for (UINT i = 0; i < devices_len; ++i) {
				if (devices[i].dwType == RIM_TYPEMOUSE) {
					have_mouse = true;
					puts("MOUSE");
				}
				if (devices[i].dwType == RIM_TYPEKEYBOARD) {
					have_keyboard = true;
					puts("KB");
				}
			}
		}
		free(devices);
	}

	return have_keyboard && have_mouse;
}

int main()
{
    printf("have? %d\n", have_keyboard_n_mouse());
}
```

```
$ sleep 3 && winegcc main.cpp -o app && wine ./app.exe
00e0:fixme:ver:GetCurrentPackageId (00007FFFFE2FFEB0 0000000000000000): stub
0034:fixme:ver:GetCurrentPackageId (00007FFFFE2FFEB0 0000000000000000): stub
devs: 2
MOUSE
KB
have? 1
0128:fixme:ver:GetCurrentPackageId (00007FFFFE2FFEB0 0000000000000000): stub
003c:fixme:ver:GetCurrentPackageId (00007FFFFE2FFEB0 0000000000000000): stub
```

Reply:
How about doing that for 5.10.0? And just removing this code for 5.9.0